src/argilla/client/feedback/dataset/remote/dataset.py

```diff
@@ -465,10 +466,58 @@ def add_metadata_property(
         # TODO(alvarobartt): structure better the mixins to be able to easily reuse those, here to avoid circular imports
         from argilla.client.feedback.dataset.mixins import ArgillaMixin
 
-        metadata_property = ArgillaMixin._parse_to_remote_metadata_property(metadata_property)
-        self._metadata_properties.append(metadata_property)
-        self._metadata_properties_mapping[metadata_property.name] = metadata_property
-        return metadata_property
+        return ArgillaMixin._parse_to_remote_metadata_property(metadata_property=metadata_property, client=self._client)
+
+    @allowed_for_roles(roles=[UserRole.owner, UserRole.admin])
+    def delete_metadata_properties(
+        self, metadata_properties: Union[str, List[str]]
+    ) -> Union["AllowedMetadataPropertyTypes", List["AllowedMetadataPropertyTypes"]]:
+        """Deletes a list of `metadata_properties` from the current `FeedbackDataset`
+        in Argilla.
+
+        Note:
+            Existing `FeedbackRecord`s if any, will remain unchanged if those contain metadata
+            named the same way as the `metadata_properties` to delete, but the validation will
+            be removed as well as `metadata_property` index, which means one won't be able to
+            use that for filtering.
+
+        Args:
+            metadata_properties: the metadata property/ies name/s to delete from the current
+                `FeedbackDataset` in Argilla.
+
+        Returns:
+            The `metadata_property` or `metadata_properties` deleted from the current
+            `FeedbackDataset` in Argilla, but using the local schema e.g. if you delete a
+            `RemoteFloatMetadataProperty` this method will delete it from Argilla and will
+            return a `FloatMetadataProperty` instance.
+
+        Raises:
+            PermissionError: if the user does not have either `owner` or `admin` role.
+            RuntimeError: if the `metadata_properties` cannot be deleted from the current
+                `FeedbackDataset` in Argilla.
+        """
+        if isinstance(metadata_properties, str):
+            metadata_properties = [metadata_properties]
+
+        # TODO(alvarobartt): structure better the mixins to be able to easily reuse those, here to avoid circular imports
+        from argilla.client.feedback.dataset.mixins import ArgillaMixin
+
+        deleted_metadata_properties = []
+        for metadata_property in metadata_properties:
+            try:
+                metadata_property = metadata_properties_api_v1.delete_metadata_property(
+                    client=self._client, id=self.id
+                ).parsed
```

Comment:
or why not just
```suggestion
                metadata_property.delete()
```

Reply:
True, I implemented first the `delete_metadata_property` method and then I forgot to change it, nice catch!